cinema/views.py

```diff
@@ -21,24 +23,49 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = [TokenAuthentication, ]
+    permission_classes = [IsAdminOrIfAuthenticatedReadOnly, ]
 
 
```

Comment:
There is a better way to do it, instead of copying code. Use REST_FRAMEWORK constant in settings.py and add DEFAULT_PERMISSION_CLASSES and DEFAULT_AUTHENTICATION_CLASSES and then override the default permissions where needed